iceberg/iceberg-handler/src/main/java/org/apache/iceberg/mr/hive/HiveIcebergMetaHook.java

```diff
@@ -310,6 +335,26 @@ public void rollbackAlterTable(org.apache.hadoop.hive.metastore.api.Table hmsTab
     }
   }
 
+  private static boolean isSupportedAlterOperation(EnvironmentContext context) {
+    for (Enum<?> op : SUPPORTED_ALTER_OPS) {
+      if (isMatchingAlterOp(op, context)) {
+        return true;
+      }
+    }
+    return false;
+  }
+
+  private static boolean isMatchingAlterOp(Enum<?> alterOperation, EnvironmentContext context) {
+    if (context == null) {
```

Comment:
these two null checks could be extracted into `isSupportedAlterOperation`, since if they're null they'll always be false so no need for looping